dashboard/app/dsl/base_dsl.rb

```diff
@@ -32,7 +32,7 @@ def self.parse(str, filename, name=nil)
     object = new
     object.name(name) if name.present?
     object.instance_eval(str.to_ascii, filename)
-    [object.parse_output, object.i18n_hash]
+    [object.parse_output.tap {|out| out[:md5] = Digest::MD5.hexdigest(str)}, object.i18n_hash]
```

Comment:
Nit: clearer without `tap`?

```suggestion
    out = object.parse_output
    out[:md5] = Digest::MD5.hexdigest(str)
    [out, object.i18n_hash]
```